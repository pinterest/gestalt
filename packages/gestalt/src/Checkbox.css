:root {
  --red: #bd081c;
  --size-sm: 16px;
  --size-md: 24px;
}

.border {
  border-color: #8e8e8e;
}

.borderError {
  border-color: #e3780c;
}

.borderDark {
  border-color: #333;
}

.check {
  composes: block from "./Layout.css";
  composes: borderBox from "./Layout.css";
  composes: flex from "./Layout.css";
  composes: itemsCenter from "./Layout.css";
  composes: justifyCenter from "./Layout.css";
  border-style: solid;
  border-width: 1px;
}

.checkEnabled {
  composes: pointer from "./Cursor.css";
}

.checkFocused {
  composes: accessibilityOutlineFocus from "./Focus.css";
}

.checkSm {
  border-radius: 3px;
  height: var(--size-sm);
  width: var(--size-sm);
}

.checkMd {
  border-radius: 4px;
  height: var(--size-md);
  width: var(--size-md);
}

.input {
  composes: absolute from "./Layout.css";
  composes: m0 from "./Whitespace.css";
  appearance: none;
  opacity: 0;
  outline: 0;
}

.inputEnabled {
  composes: pointer from "./Cursor.css";
}

.inputSm {
  height: var(--size-sm);
  width: var(--size-sm);
}

.inputMd {
  height: var(--size-md);
  width: var(--size-md);
}
