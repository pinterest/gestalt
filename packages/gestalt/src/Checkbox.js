// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import colors from './Colors.css';
import styles from './Checkbox.css';
import Box from './Box.js';
import Icon from './Icon.js';

type Props = {|
  checked?: boolean,
  disabled?: boolean,
  hasError?: boolean,
  id: string,
  indeterminate?: boolean,
  name?: string,
  onChange: ({ event: SyntheticInputEvent<>, checked: boolean }) => void,
  onClick?: ({
    event: SyntheticInputEvent<HTMLInputElement>,
    checked: boolean,
  }) => void,
  size?: 'sm' | 'md',
|};

type State = {|
  focused: boolean,
|};

export default class Checkbox extends React.Component<Props, State> {
  static propTypes = {
    checked: PropTypes.bool,
    disabled: PropTypes.bool,
    hasError: PropTypes.bool,
    id: PropTypes.string.isRequired,
    indeterminate: PropTypes.bool,
    name: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    onClick: PropTypes.func,
    size: PropTypes.oneOf(['sm', 'md']),
  };

  static defaultProps = {
    checked: false,
    disabled: false,
    hasError: false,
    indeterminate: false,
    size: 'md',
  };

  state = {
    focused: false,
  };

  componentDidMount() {
    if (this.props.indeterminate) {
      this.setIndeterminate(!!this.props.indeterminate);
    }
  }

  componentDidUpdate(previousProps: Props) {
    if (previousProps.indeterminate !== this.props.indeterminate) {
      this.setIndeterminate(!!this.props.indeterminate);
    }
  }

  setIndeterminate(indeterminate: boolean) {
    if (this.input) {
      this.input.indeterminate = indeterminate;
    }
  }

  handleChange = (event: SyntheticInputEvent<>) => {
    const { checked } = event.target;
    this.props.onChange({ event, checked });
  };

  handleClick = (event: SyntheticInputEvent<HTMLInputElement>) => {
    const { onClick } = this.props;
    if (onClick) {
      const { checked } = event.currentTarget;
      onClick({ event, checked });
    }
  };

  handleBlur = () => this.setState({ focused: false });

  handleFocus = () => this.setState({ focused: true });

  input: ?HTMLInputElement;

  render() {
    const {
      checked,
      disabled,
      hasError,
      id,
      indeterminate,
      name,
      size,
    } = this.props;

    let borderStyle = styles.border;
    if (!disabled && (checked || indeterminate)) {
      borderStyle = styles.borderDark;
    } else if (hasError) {
      borderStyle = styles.borderError;
    }

    return (
      <Box position="relative">
        <input
          checked={checked}
          className={classnames(styles.input, {
            [styles.inputEnabled]: !disabled,
            [styles.indeterminate]: indeterminate,
            [styles.inputSm]: size === 'sm',
            [styles.inputMd]: size === 'md',
          })}
          disabled={disabled}
          id={id}
          name={name}
          onBlur={this.handleBlur}
          onChange={this.handleChange}
          onClick={this.handleClick}
          onFocus={this.handleFocus}
          ref={el => {
            this.input = el;
          }}
          type="checkbox"
        />
        <div
          className={classnames(
            borderStyle,
            styles.check,
            // eslint-disable-next-line no-nested-ternary
            disabled
              ? checked || indeterminate
                ? colors.grayBg
                : colors.lightGrayBg
              : checked || indeterminate
                ? colors.darkGrayBg
                : colors.whiteBg,
            {
              [styles.checkEnabled]: !disabled,
              [styles.checkFocused]: this.state.focused,
              [styles.checkMd]: size === 'md',
              [styles.checkSm]: size === 'sm',
            }
          )}
        >
          {(checked || indeterminate) && (
            <Icon
              accessibilityLabel=""
              color="white"
              icon={indeterminate ? 'dash' : 'check'}
              size={size === 'sm' ? 8 : 12}
            />
          )}
        </div>
      </Box>
    );
  }
}
