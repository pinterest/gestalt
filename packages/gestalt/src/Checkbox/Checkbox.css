:root {
  --red: #bd081c;
  --size-sm: 16px;
  --size-md: 24px;
}

.check {
  composes: block from "../Layout.css";
  composes: borderBox from "../Layout.css";
  composes: flex from "../Layout.css";
  composes: itemsCenter from "../Layout.css";
  composes: justifyCenter from "../Layout.css";
  border-style: solid;
  border-width: 1px;
}

.checkEnabled {
  composes: pointer from "../Cursor.css";
}

.checkDarkGray {
  composes: darkGrayBg from "../Colors.css";
  border-color: #555;
}

.checkGray {
  composes: grayBg from "../Colors.css";
  border-color: #b5b5b5;
}

.checkLightGray {
  composes: lightGrayBg from "../Colors.css";
  border-color: #b5b5b5;
}

.checkWhite {
  composes: whiteBg from "../Colors.css";
  border-color: #b5b5b5;
}

.checkFocused {
  composes: accessibilityOutlineFocus from "../Focus.css";
}

.checkSm {
  border-radius: 3px;
  height: var(--size-sm);
  width: var(--size-sm);
}

.checkMd {
  border-radius: 4px;
  height: var(--size-md);
  width: var(--size-md);
}

.input {
  composes: absolute from "../Layout.css";
  composes: m0 from "../Whitespace.css";
  appearance: none;
  opacity: 0;
  outline: 0;
}

.inputEnabled {
  composes: pointer from "../Cursor.css";
}

.inputSm {
  height: var(--size-sm);
  width: var(--size-sm);
}

.inputMd {
  height: var(--size-md);
  width: var(--size-md);
}
