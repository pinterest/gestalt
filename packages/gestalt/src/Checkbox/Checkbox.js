// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './Checkbox.css';
import Box from '../Box/Box';
import Icon from '../Icon/Icon';

type Props = {|
  checked?: boolean,
  disabled?: boolean,
  id: string,
  indeterminate?: boolean,
  name?: string,
  onChange: ({ event: SyntheticInputEvent<>, checked: boolean }) => void,
  size?: 'sm' | 'md',
|};

type State = {|
  focused: boolean,
|};

export default class Checkbox extends React.Component<Props, State> {
  static propTypes = {
    checked: PropTypes.bool,
    disabled: PropTypes.bool,
    id: PropTypes.string.isRequired,
    indeterminate: PropTypes.bool,
    name: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    size: PropTypes.oneOf(['sm', 'md']),
  };

  static defaultProps = {
    checked: false,
    disabled: false,
    indeterminate: false,
    size: 'md',
  };

  state: State = {
    focused: false,
  };

  componentDidMount() {
    if (this.props.indeterminate) {
      this.setIndeterminate(!!this.props.indeterminate);
    }
  }

  componentDidUpdate(previousProps: Props) {
    if (previousProps.indeterminate !== this.props.indeterminate) {
      this.setIndeterminate(!!this.props.indeterminate);
    }
  }

  setIndeterminate(indeterminate: boolean) {
    if (this.input) {
      this.input.indeterminate = indeterminate;
    }
  }

  input: ?HTMLInputElement;

  handleChange = (event: SyntheticInputEvent<>) => {
    const { checked } = event.target;
    this.props.onChange({ event, checked });
  };

  handleBlur = () => this.setState({ focused: false });
  handleFocus = () => this.setState({ focused: true });

  render() {
    const { checked, disabled, id, indeterminate, name, size } = this.props;
    return (
      <Box position="relative">
        <input
          checked={checked}
          className={classnames(styles.input, {
            [styles.inputEnabled]: !disabled,
            [styles.indeterminate]: indeterminate,
            [styles.inputSm]: size === 'sm',
            [styles.inputMd]: size === 'md',
          })}
          disabled={disabled}
          id={id}
          name={name}
          onBlur={this.handleBlur}
          onChange={this.handleChange}
          onFocus={this.handleFocus}
          ref={el => {
            this.input = el;
          }}
          type="checkbox"
        />
        <div
          className={classnames(
            styles.check,
            // eslint-disable-next-line no-nested-ternary
            disabled
              ? checked || indeterminate
                ? styles.checkGray
                : styles.checkLightGray
              : checked || indeterminate
                ? styles.checkDarkGray
                : styles.checkWhite,
            {
              [styles.checkEnabled]: !disabled,
              [styles.checkFocused]: this.state.focused,
              [styles.checkMd]: size === 'md',
              [styles.checkSm]: size === 'sm',
            }
          )}
        >
          {(checked || indeterminate) && (
            <Icon
              accessibilityLabel=""
              color="white"
              icon={indeterminate ? 'dash' : 'check'}
              size={size === 'sm' ? 8 : 12}
            />
          )}
        </div>
      </Box>
    );
  }
}
