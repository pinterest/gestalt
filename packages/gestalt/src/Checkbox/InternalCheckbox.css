:root {
  --g-size-sm: var(--space-400);
  --g-size-md: var(--space-600);
}

.sizeSm {
  height: var(--g-size-sm);
  width: var(--g-size-sm);
}

.sizeMd {
  height: var(--g-size-md);
  width: var(--g-size-md);
}

.enabled {
  background-color: var(--color-background-checkbox-unchecked-default);
}

.disabled {
  background-color: var(--color-background-checkbox-disabled);
}

.checked {
  background-color: var(--color-background-checkbox-checked-default);
}

.errorChecked {
  background-color: var(--color-background-checkbox-checked-error);
}

.error {
  background-color: var(--color-background-checkbox-unchecked-error);
}

.border {
  border-color: var(--color-border-checkbox-unchecked-default);
}

.borderSelected {
  border-color: var(--color-border-checkbox-checked-default);
}

.borderDisabled {
  border-color: var(--color-border-checkbox-disabled);
}

.borderErrorChecked {
  border-color: var(--color-border-checkbox-checked-error);
}

.borderErrorUnchecked {
  border-color: var(--color-border-checkbox-unchecked-error);
}

.borderHovered {
  border-color: var(--color-border-checkbox-unchecked-hover);
}

.borderHoveredError {
  border-color: var(--sema-color-hover-border-error);
}

.borderPressed {
  border-color: var(--sema-color-pressed-border-interactive);
}

.borderPressedError {
  border-color: var(--sema-color-pressed-border-error);
}

.hovered {
  background-color: var(--sema-color-hover-background-selected);
  border-color: var(--sema-color-hover-background-selected);
}

.focus {
  border-color: var(--sema-color-border-focus-inner-default);
  border-style: solid;
  border-width: 2px;
  outline: 2px solid var(--sema-color-border-focus-outer-default);
  outline-offset: 0;
}

.hoveredError {
  background-color: var(--sema-color-hover-background-error-weak);
  border-color: var(--sema-color-hover-background-error-weak);
}

.pressed {
  background-color: var(--sema-color-pressed-background-selected);
  border-color: var(--sema-color-pressed-background-selected);
}

.pressedError {
  background-color: var(--sema-color-pressed-background-error-weak);
  border-color: var(--sema-color-pressed-background-error-weak);
}

.borderRadiusSm {
  composes: rounding1 from "../Borders.css";
}

.borderRadiusMd {
  composes: rounding2 from "../Borders.css";
}

.vrBorderRadiusSm {
  border-radius: var(--sema-rounding-xs);
}

.vrBorderRadiusMd {
  border-radius: var(--sema-rounding-sm);
}

.check {
  composes: block from "../Layout.css";
  composes: borderBox from "../Layout.css";
  composes: flex from "../Layout.css";
  composes: xsItemsCenter from "../Layout.css";
  composes: justifyCenter from "../Layout.css";
  composes: solid from "../Borders.css";
}

.sizeLg {
  height: var(--space-600);
  width: var(--space-600);
}

.thinBorder {
  composes: sizeSm from "../Borders.css";
}

.thickBorder {
  composes: sizeLg from "../Borders.css";
}

.inputEnabled {
  composes: pointer from "../Cursor.css";
}

.readOnly {
  pointer-events: none;
}

.input {
  composes: absolute from "../Layout.css";
  composes: m0 from "../Whitespace.css";
  appearance: none;
  opacity: var(--opacity-0);
  outline: 0;
}
