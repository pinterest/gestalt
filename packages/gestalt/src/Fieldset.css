.marginTop {
  margin-top: var(--sema-space-300);
}
