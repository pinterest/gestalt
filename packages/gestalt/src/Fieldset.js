// @flow strict
import { type Node } from 'react';
import classnames from 'classnames';
import boxStyles from './Box.css';
import boxWhitespaceStyles from './boxWhitespace.css';
import labelStyles from './Label/InternalLabel.css';
import formStyles from './shared/FormElement.css';
import FormErrorMessage from './shared/FormErrorMessage.js';
import formLabelStyles from './shared/FormLabel.css';
import Text from './Text.js';
import whitespaceStyles from './Whitespace.css';

type Props = {
  /**
   * The content of Fieldset, typically [RadioButtons](https://gestalt.pinterest.systems/web/radiobutton), [Checkboxes](https://gestalt.pinterest.systems/web/checkbox) or [TextFields](https://gestalt.pinterest.systems/web/textfield).
   */
  children: Node,
  /**
   * A unique identifier for this Fieldset. `id` must be specified when an errorMessage is added.
   */
  id?: string,
  /**
   * When needed, pass a string with a helpful error message (be sure to localize!).
   */
  errorMessage?: string,
  /**
   * Caption that clearly and concisely describes the form elements grouped in the fieldset.
   */
  legend: string,
  /**
   * Whether the legend should be visible or not. If `hidden`, the legend is still available for screen reader users, but does not appear visually. See the [legend visibility variant](https://gestalt.pinterest.systems#Legend-visibility) for more info.
   */
  legendDisplay?: 'visible' | 'hidden',
};

/**
 * [Fieldset](https://gestalt.pinterest.systems/web/fieldset) creates a fieldset and legend for a group of related form items, like [RadioButtons](https://gestalt.pinterest.systems/web/radiobutton) or [CheckBoxes](https://gestalt.pinterest.systems/web/checkbox), in order to clearly indicate related form items."
 *
 * ![Fieldset light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/Fieldset.spec.mjs-snapshots/Fieldset-chromium-darwin.png)
 * ![Fieldset dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/Fieldset-dark.spec.mjs-snapshots/Fieldset-dark-chromium-darwin.png)
 *
 */
export default function Fieldset({
  id = '',
  errorMessage,
  legend,
  legendDisplay = 'visible',
  children,
}: Props): Node {
  if (errorMessage && id === '') {
    // eslint-disable-next-line no-console
    console.error('Please provide an id property to <Fieldset />');
  }

  return (
    <fieldset
      className={classnames(formStyles.unstyled, whitespaceStyles.p0, whitespaceStyles.m0)}
      aria-describedby={errorMessage ? `${id}-error` : undefined}
    >
      <legend
        className={classnames(
          labelStyles.label,
          formLabelStyles.formLabel,
          boxWhitespaceStyles.paddingX0, // Needed to remove the default legend  padding
          {
            [boxStyles.visuallyHidden]: legendDisplay === 'hidden',
          },
        )}
      >
        <Text size="100">{legend}</Text>
      </legend>
      {children}
      {errorMessage && <FormErrorMessage id={`${id}-error`} text={errorMessage} />}
    </fieldset>
  );
}
