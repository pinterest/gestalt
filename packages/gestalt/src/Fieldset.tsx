import { ReactNode } from 'react';
import classnames from 'classnames';
import boxStyles from './Box.css';
import boxWhitespaceStyles from './boxWhitespace.css';
import labelStyles from './Label/InternalLabel.css';
import formStyles from './sharedSubcomponents/FormElement.css';
import FormErrorMessage from './sharedSubcomponents/FormErrorMessage';
import formLabelStyles from './sharedSubcomponents/FormLabel.css';
import Text from './Text';
import useInExperiment from './useInExperiment';
import whitespaceStyles from './Whitespace.css';

type Props = {
  /**
   * The content of Fieldset, typically [RadioButtons](https://gestalt.pinterest.systems/web/radiobutton), [Checkboxes](https://gestalt.pinterest.systems/web/checkbox) or [TextFields](https://gestalt.pinterest.systems/web/textfield).
   */
  children: ReactNode;
  /**
   * A unique identifier for this Fieldset. `id` must be specified when an errorMessage is added.
   */
  id?: string;
  /**
   * When needed, pass a string with a helpful error message (be sure to localize!).
   */
  errorMessage?: string;
  /**
   * Caption that clearly and concisely describes the form elements grouped in the fieldset.
   */
  legend: string;
  /**
   * Whether the legend should be visible or not. If `hidden`, the legend is still available for screen reader users, but does not appear visually. See the [legend visibility variant](https://gestalt.pinterest.systems#Legend-visibility) for more info.
   */
  legendDisplay?: 'visible' | 'hidden';
};

/**
 * [Fieldset](https://gestalt.pinterest.systems/web/fieldset) creates a fieldset and legend for a group of related form items, like [RadioButtons](https://gestalt.pinterest.systems/web/radiobutton) or [CheckBoxes](https://gestalt.pinterest.systems/web/checkbox), in order to clearly indicate related form items."
 *
 * ![Fieldset light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/Fieldset.spec.ts-snapshots/Fieldset-chromium-darwin.png)
 * ![Fieldset dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/Fieldset-dark.spec.ts-snapshots/Fieldset-dark-chromium-darwin.png)
 *
 */
export default function Fieldset({
  id = '',
  errorMessage,
  legend,
  legendDisplay = 'visible',
  children,
}: Props) {
  const isInVRExperiment = useInExperiment({
    webExperimentName: 'web_gestalt_visualRefresh',
    mwebExperimentName: 'web_gestalt_visualRefresh',
  });

  if (errorMessage && id === '') {
    // eslint-disable-next-line no-console
    console.error('Please provide an id property to <Fieldset />');
  }

  return (
    <fieldset
      aria-describedby={errorMessage ? `${id}-error` : undefined}
      className={classnames(formStyles.unstyled, whitespaceStyles.p0, whitespaceStyles.m0)}
    >
      <legend
        className={classnames(
          labelStyles.label,
          boxWhitespaceStyles.paddingX0, // Needed to remove the default legend  padding
          {
            [boxStyles.visuallyHidden]: legendDisplay === 'hidden',
            [formLabelStyles.formLabel]: !isInVRExperiment,
            [formLabelStyles.vrFormLabel]: isInVRExperiment,
          },
        )}
      >
        <Text size="100">{legend}</Text>
      </legend>
      {children}
      {errorMessage && <FormErrorMessage id={`${id}-error`} text={errorMessage} />}
    </fieldset>
  );
}

Fieldset.displayName = 'Fieldset';
