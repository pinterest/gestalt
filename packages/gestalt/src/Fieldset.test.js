// @flow strict
import { create } from 'react-test-renderer';
import Checkbox from './Checkbox';
import Fieldset from './Fieldset';

describe('Fieldset', () => {
  it('renders', () => {
    const tree = create(
      <Fieldset legend="What is your favorite dog?">
        <Checkbox id="Schnauzer" label="Schnauzer" onChange={() => {}} />
        <Checkbox id="Aussie" label="Aussie" onChange={() => {}} />
      </Fieldset>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a visually hidden legend', () => {
    const tree = create(
      <Fieldset legend="What is your favorite dog?" legendDisplay="hidden">
        <Checkbox id="Schnauzer" label="Schnauzer" onChange={() => {}} />
        <Checkbox id="Aussie" label="Aussie" onChange={() => {}} />
      </Fieldset>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with errorMessage', () => {
    const tree = create(
      <Fieldset
        errorMessage="At least 1 item must be selected"
        id="fieldset-with-error"
        legend="What is your favorite dog?"
        legendDisplay="hidden"
      >
        <Checkbox id="Schnauzer" label="Schnauzer" onChange={() => {}} />
        <Checkbox id="Aussie" label="Aussie" onChange={() => {}} />
      </Fieldset>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
