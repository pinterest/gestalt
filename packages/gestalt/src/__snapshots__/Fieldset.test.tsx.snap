// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fieldset renders 1`] = `
<fieldset
  className="unstyled p0 m0"
>
  <legend
    className="label formLabel paddingX0"
  >
    <div
      className="Text fontSize100 default alignStart breakWord fontWeightNormal"
    >
      What is your favorite dog?
    </div>
  </legend>
  <div
    className="box"
  >
    <div
      className="box marginEndN1 marginStartN1 xsDisplayFlex xsItemsStart"
    >
      <div
        className="box paddingX1 relative"
      >
        <input
          aria-invalid="false"
          checked={false}
          className="input sizeMd inputEnabled"
          disabled={false}
          id="Schnauzer"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          type="checkbox"
        />
        <div
          className="enabled border borderRadiusMd sizeMd check thickBorder"
        />
      </div>
      <div
        className="box xsDisplayBlock"
        style={
          Object {
            "marginTop": "2px",
          }
        }
      >
        <label
          className="label"
          htmlFor="Schnauzer"
        >
          <div
            className="box paddingX1"
          >
            <div
              className="Text fontSize300 default alignStart breakWord fontWeightNormal"
            >
              Schnauzer
            </div>
          </div>
        </label>
        <div
          className="box paddingX1"
        />
      </div>
    </div>
  </div>
  <div
    className="box"
  >
    <div
      className="box marginEndN1 marginStartN1 xsDisplayFlex xsItemsStart"
    >
      <div
        className="box paddingX1 relative"
      >
        <input
          aria-invalid="false"
          checked={false}
          className="input sizeMd inputEnabled"
          disabled={false}
          id="Aussie"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          type="checkbox"
        />
        <div
          className="enabled border borderRadiusMd sizeMd check thickBorder"
        />
      </div>
      <div
        className="box xsDisplayBlock"
        style={
          Object {
            "marginTop": "2px",
          }
        }
      >
        <label
          className="label"
          htmlFor="Aussie"
        >
          <div
            className="box paddingX1"
          >
            <div
              className="Text fontSize300 default alignStart breakWord fontWeightNormal"
            >
              Aussie
            </div>
          </div>
        </label>
        <div
          className="box paddingX1"
        />
      </div>
    </div>
  </div>
</fieldset>
`;

exports[`Fieldset renders a visually hidden legend 1`] = `
<fieldset
  className="unstyled p0 m0"
>
  <legend
    className="label formLabel paddingX0 visuallyHidden"
  >
    <div
      className="Text fontSize100 default alignStart breakWord fontWeightNormal"
    >
      What is your favorite dog?
    </div>
  </legend>
  <div
    className="box"
  >
    <div
      className="box marginEndN1 marginStartN1 xsDisplayFlex xsItemsStart"
    >
      <div
        className="box paddingX1 relative"
      >
        <input
          aria-invalid="false"
          checked={false}
          className="input sizeMd inputEnabled"
          disabled={false}
          id="Schnauzer"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          type="checkbox"
        />
        <div
          className="enabled border borderRadiusMd sizeMd check thickBorder"
        />
      </div>
      <div
        className="box xsDisplayBlock"
        style={
          Object {
            "marginTop": "2px",
          }
        }
      >
        <label
          className="label"
          htmlFor="Schnauzer"
        >
          <div
            className="box paddingX1"
          >
            <div
              className="Text fontSize300 default alignStart breakWord fontWeightNormal"
            >
              Schnauzer
            </div>
          </div>
        </label>
        <div
          className="box paddingX1"
        />
      </div>
    </div>
  </div>
  <div
    className="box"
  >
    <div
      className="box marginEndN1 marginStartN1 xsDisplayFlex xsItemsStart"
    >
      <div
        className="box paddingX1 relative"
      >
        <input
          aria-invalid="false"
          checked={false}
          className="input sizeMd inputEnabled"
          disabled={false}
          id="Aussie"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          type="checkbox"
        />
        <div
          className="enabled border borderRadiusMd sizeMd check thickBorder"
        />
      </div>
      <div
        className="box xsDisplayBlock"
        style={
          Object {
            "marginTop": "2px",
          }
        }
      >
        <label
          className="label"
          htmlFor="Aussie"
        >
          <div
            className="box paddingX1"
          >
            <div
              className="Text fontSize300 default alignStart breakWord fontWeightNormal"
            >
              Aussie
            </div>
          </div>
        </label>
        <div
          className="box paddingX1"
        />
      </div>
    </div>
  </div>
</fieldset>
`;

exports[`Fieldset renders with errorMessage 1`] = `
<fieldset
  aria-describedby="fieldset-with-error-error"
  className="unstyled p0 m0"
>
  <legend
    className="label formLabel paddingX0 visuallyHidden"
  >
    <div
      className="Text fontSize100 default alignStart breakWord fontWeightNormal"
    >
      What is your favorite dog?
    </div>
  </legend>
  <div
    className="box"
  >
    <div
      className="box marginEndN1 marginStartN1 xsDisplayFlex xsItemsStart"
    >
      <div
        className="box paddingX1 relative"
      >
        <input
          aria-invalid="false"
          checked={false}
          className="input sizeMd inputEnabled"
          disabled={false}
          id="Schnauzer"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          type="checkbox"
        />
        <div
          className="enabled border borderRadiusMd sizeMd check thickBorder"
        />
      </div>
      <div
        className="box xsDisplayBlock"
        style={
          Object {
            "marginTop": "2px",
          }
        }
      >
        <label
          className="label"
          htmlFor="Schnauzer"
        >
          <div
            className="box paddingX1"
          >
            <div
              className="Text fontSize300 default alignStart breakWord fontWeightNormal"
            >
              Schnauzer
            </div>
          </div>
        </label>
        <div
          className="box paddingX1"
        />
      </div>
    </div>
  </div>
  <div
    className="box"
  >
    <div
      className="box marginEndN1 marginStartN1 xsDisplayFlex xsItemsStart"
    >
      <div
        className="box paddingX1 relative"
      >
        <input
          aria-invalid="false"
          checked={false}
          className="input sizeMd inputEnabled"
          disabled={false}
          id="Aussie"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          type="checkbox"
        />
        <div
          className="enabled border borderRadiusMd sizeMd check thickBorder"
        />
      </div>
      <div
        className="box xsDisplayBlock"
        style={
          Object {
            "marginTop": "2px",
          }
        }
      >
        <label
          className="label"
          htmlFor="Aussie"
        >
          <div
            className="box paddingX1"
          >
            <div
              className="Text fontSize300 default alignStart breakWord fontWeightNormal"
            >
              Aussie
            </div>
          </div>
        </label>
        <div
          className="box paddingX1"
        />
      </div>
    </div>
  </div>
  <div
    className=""
    id="fieldset-with-error-error"
  >
    <div
      className="Text fontSize100 error alignStart breakWord fontWeightNormal"
    >
      <span
        className="formErrorMessage"
        id="fieldset-with-error-error"
      >
        <div
          className="box"
          role="alert"
        >
          <div
            className="Flex rowGap2 columnGap2 xsDirectionRow xsItemsCenter"
          >
            <div
              className="FlexItem"
            >
              <div
                className="box"
              >
                <svg
                  aria-hidden={true}
                  aria-label=""
                  className="error icon iconBlock"
                  height={16}
                  role="img"
                  viewBox="0 0 24 24"
                  width={16}
                >
                  <path
                    d="test-file-stub"
                  />
                </svg>
              </div>
            </div>
            <div
              className="FlexItem"
            >
              At least 1 item must be selected
            </div>
          </div>
        </div>
      </span>
    </div>
  </div>
</fieldset>
`;
